h2o-py/h2o/automl/_base.py

```diff
@@ -125,3 +125,72 @@ def event_log(self):
         :return: an H2OFrame with detailed events occurred during the AutoML training.
         """
         pass
+
+    def get_best_model(self, algorithm, criterion=None):
+        """
+        Get best model of a given family/algorithm from an AutoML object.
+
+        :param algorithm: One of "base_model", "deep_learning", "drf", "gbm", "glm", "stacked_ensemble", "xgboost", "xrt"
+        :param criterion: Criterium can be one of the metrics reported in leaderboard, if None pick the first metric
+                          for each task from the following list:
+                            * Regression metrics: mean_residual_deviance, rmse, mse, mae, rmsle
+                            * Binomial metrics: auc, logloss, aucpr, mean_per_class_error, rmse, mse
+                            * Multinomial metrics: mean_per_class_error, logloss, rmse, mse, auc, aucpr
+        :return: a model or None if none of a given family is present
+        :examples:
+        >>> # Set up an H2OAutoML object
+        >>> aml = H2OAutoML(max_runtime_secs=30)
+        >>> # Launch an AutoML run
+        >>> aml.train(y=y, training_frame=train)
+        >>> gbm = aml.get_best_model("gbm")
+        """
+        def _best(leaderboard, pattern, criterion, ascending):
+            # type: (h2o.H2OFrame, str, str, bool) -> h2o.model.ModelBase
+            matches = leaderboard.sort(by=criterion, ascending=ascending)["model_id"].grep(pattern)
+            if matches.nrow == 0:
+                return None
+            return h2o.get_model(self.leaderboard[int(matches[0, :]), "model_id"])
+
+        patterns = dict(
+            base_model="^(?!StackedEnsemble_)",
+            deep_learning="^DeepLearning_",
+            drf="^(DRF|XRT)_",
+            gbm="^GBM_",
+            glm="^GLM_",
+            stacked_ensemble="^StackedEnsemble_",
+            xgboost="^XGBoost_",
+            xrt="^XRT_"
+        )
```

Comment:
Please don't use those unreliable pseudo naming conventions for production code, I think it would make more sense to add something like a `type` or `algo` column to the leaderboard on the backend:
it's just a matter of registering a new `LeaderboardCell` in `AutoML.createLeaderboardExtensionProvider`, then you just use the extended leaderboard:
```
lb = get_leaderboard(self, ['type'])
```

Reply:
Great suggestion, thanks!